kernel/src/hil/digest.rs

```diff
@@ -0,0 +1,48 @@
+//! Interface for Digest
+
+use crate::returncode::ReturnCode;
+
+/// Implement this trait and use `set_client()` in order to receive callbacks.
+pub trait Client<'a> {
+    /// On error `result` will contain device specific error codes
+    fn hash_done(&'a self, result: Result<(), u32>);
```

Comment:
A HIL should abstract the HW. Having device specific error codes doesn't gain anything. It'd be better to have a mapping of error scenarios to an enum such as `ReturnCode`, but I don't yet know of any convention here. I always choose a variant which _feels right_.

In line with my other comment, this should provide the calculated hash.

Reply:
I would prefer to have specific errors, the problem is that I don't have a list of possible OpenTitan errors, and even if I did I don't have a list of other module errors. So it seems hard to generalise this to a small list of errors. My thinking is that during operation we aren't really expecting errors (what can fail?) and that this is just for debugging.

Reply:
Still, this is a HIL. This way, it would be very incompatible with any other HW being interfaced with. If you can't find a list, and don't know what could go wrong, I'd suggest always returning `ReturnCode::FAIL` if there has been an error - the client won't know how to interpret the `u32` either.